lib/viewmodels/create_accout_view_model.dart

```diff
@@ -0,0 +1,120 @@
+import "dart:async";
+
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/user/user_data.dart";
+import "package:proxima/models/ui/create_account_errors.dart";
+import "package:proxima/services/database/user_repository_service.dart";
+import "package:proxima/viewmodels/login_view_model.dart";
+
+class CreateAccountViewModel extends AsyncNotifier<CreateAccountErrors?> {
```

Comment:
Very good job with the view model, you perform a strong validation and the code is well documented which made it easy for me to understand :+1: 

I just believe that we don't really need to have the `CreateAccountErrors` nullable for the `AsyncNotifier`.
Conceptually, I think that having it `null` would represent the same state as:
```dart
  const CreateAccountErrors({
    uniqueUsernameError = null,
    pseudoError = null,
    created = false,
  });
```

Reply:
Indeed, there was a mixing in my reasoning because of the original variable `valid`, which rendered the nullable state superfluous. 
I think you are correct about the fact it simplifies the code to have it non-nullable, so fixed in b0f4eb7cf45a2264956f0014196e1d89abe80789.